### PR TITLE
Fix wrap-ssl-redirect calling handler on redirect

### PR DESCRIPTION
**Pull request by mainej**
Commit a49582804 introduced a bug in which the handler passed to
wrap-ssl-redirect was always called, regardless of whether the request
was over SSL. This is a bit like launching the missles, then saying "I'm
sorry I can't launch, you have the wrong code."

This bug wasn't caught by the tests because, until now none of them
checked whether the inner handler is called.

### CODE REVIEW

**Comment by weavejester** [comment]
Thanks for the fix!

Could you possibly add a docstring to `ssl-redirect`, perhaps something like:

```clojure
  "Given a HTTP request, return a redirect response to the equivalent HTTPS URL.
  See: wrap-ssl-redirect."
```

Also, could you change the commit message to:

```
Fix wrap-ssl-redirect calling handler on redirect

Handler should not be called when a redirect response is returned.
```

This explains the issue more concisely, and sticks to imperative mood in the summary line.

File: test/ring/middleware/ssl_test.clj [hunk 1]
```diff
@@ -75,14 +75,25 @@
...
+    (testing "HTTP POST request doesn't cause side-effects"
+      (handler (request :post "/"))
+      (is (not (realized? effect))))))
 
 (deftest test-wrap-ssl-redirect-cps
   (testing "HTTP GET request"
-    (let [handler (wrap-ssl-redirect (fn [_ respond _] (respond (response ""))))
+    (let [effect  (promise)
+          handler (wrap-ssl-redirect (fn [_ respond _] (respond (do (effect "performed")
+                                                                   (response "")))))
```
Comment (weavejester) [inline test/ring/middleware/ssl_test.clj]: Indentation needs fixing, here?

**Comment by mainej** [comment]
Yep, good suggestions. I'll push those changes shortly. 

Also, should the `ssl-redirect` function be named `ssl-redirect-response`? I'm not sure of the convention here. And finally, should the shared conditional be extracted, something like `ssl-request?`

Thanks for writing and maintaining so many useful Clojure libraries!

**Comment by weavejester** [comment]
The convention I'm using is that if I have a middleware function `wrap-foo`, then the "request" part of it should be handled by a function `foo-request`, and the response part handled by `foo-response`.

In this case it only makes sense to call the response part if the request is HTTP, so... yes, on reflection I think `ssl-redirect` should be renamed to `ssl-redirect-response`.

**Comment by mainej** [comment]
`ssl-redirect` renamed to `ssl-redirect-response`

**Pull request merged**